src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -113,9 +117,30 @@ private void Init(string pattern, RegexOptions options, TimeSpan matchTimeout, C
             caps = _code.Caps;
             capsize = _code.CapSize;
 
+            // if SRM is used then construct the SMR.Regex matcher
+            // this construction fails and throws a NotSupportedException
+            // for some unsupported constructs used in the original regex
+            _useSRM = (options & RegexOptions.DFA) != 0;
+            if (_useSRM)
+                // remove the DFA flag because it is undefined in SRM
+                _srm = InitializeSRM(tree.Root, options & ~RegexOptions.DFA);
+
             InitializeReferences();
         }
 
+        /// <summary>
+        /// This method is here for perf reasons: if InitializeSRM is NOT called in the
+        /// Init method, we don't load SRM.Regex when instantiating a regex that does not use the DFA option
```

Comment:
Now it's all in the same assembly, does no-inlining matter? The type will be loaded anyway, and Jitting will only happen if it ends up needing the code.

Reply:
Removed the comment and the complier method attribute. Added another note there as a reminder that in fact RightToLeft can be supported now -- fairly easily I think.